opencodelists/tests/fixtures.py

```diff
@@ -0,0 +1,289 @@
+
+    # version_with_complete_searches
+    # - belongs to new_style_codelist
+    # - has some searches, and all codes covered
+    version_with_complete_searches = export_to_builder(
+        version=version_with_some_searches, owner=organisation_user
+    )
+    create_search(
+        draft=version_with_complete_searches,
+        term="elbow",
+        codes=codes_for_search_term("elbow"),
+    )
+    update_code_statuses(
+        draft=version_with_complete_searches,
+        updates=[("116309007", "-")],  # exclude "Finding of elbow region"
+    )
+    save(draft=version_with_complete_searches)
+
+    # draft_codelist
+    # - belongs to organisation
+    # - has single version, being edited:
+    #   - draft_version
+    draft_codelist = create_codelist_from_scratch(
+        owner=organisation,
+        name="Draft Codelist",
+        coding_system_id="snomedct",
+        draft_owner=organisation_user,
+    )
+
+    # draft_version
+    # - belongs to draft_codelist
+    # - being edited by organisation_user
+    draft_version = draft_codelist.versions.get()
+
+    # user_codelist
+    # - belongs to organisation_user
+    # - has one version:
+    #   - user_version
+    user_codelist = create_codelist_with_codes(
+        owner=organisation_user,
+        name="User-owned Codelist",
+        coding_system_id="snomedct",
+        codes=load_codes_from_csv("disorder-of-elbow-excl-arthritis.csv"),
+    )
+
+    # user_version
+    # - belongs to user_codelist
+    user_version = user_codelist.versions.get()
+
+    assert set(locals()) == AVAILABLE_FIXTURES
+    return locals()
+
+
+def load_csv_data(filename):
+    """Return CSV data in given filename."""
+
+    with open(SNOMED_FIXTURES_PATH / filename) as f:
+        return f.read()
+
+
+def load_codes_from_csv(filename):
+    """Return codes in CSV file at given filename."""
+
+    with open(SNOMED_FIXTURES_PATH / filename) as f:
+        rows = list(csv.reader(f))
+
+    return [row[0] for row in rows[1:]]
+
+
+def codes_for_search_term(term):
+    """Return codes matching search term."""
+
+    coding_system = CODING_SYSTEMS["snomedct"]
+    return do_search(coding_system, term)["all_codes"]
+
+
+# Create each fixture in the module scope.
+for fixture_name in AVAILABLE_FIXTURES:
+
+    def build_fixture(fixture_name):
+        """Build a fixture function."""
+
+        def fixture(universe):
+            """The actual pytest fixture.
+
+            Finds the member of the universe with the given name, reloads it from the
+            database, and returns it.
+            """
+            obj = universe[fixture_name]
+            obj.refresh_from_db()
+            return obj
+
+        return fixture
+
+    # Update the module's locals with the fixture's name.  This lets us do:
+    #
+    #   from opencodelists.tests.fixtures import *
+    #
+    # in conftest.py.
+    locals()[fixture_name] = pytest.fixture(scope="function")(
+        build_fixture(fixture_name)
+    )
```

Comment:
Maybe I'm missing something here, but why not build them all in a `conftest.py` module to save doing this bit?